source/api/commands/getcookie.md

```diff
@@ -63,8 +63,43 @@ In this example, on first login, our server sends us back a session cookie.
 // assume we just logged in
 cy.contains('Login').click()
 cy.url().should('include', 'profile')
+// retries until cookie with value=189jd09su
+// is found or default command timeout ends
 cy.getCookie('session_id')
   .should('have.property', 'value', '189jd09su')
+  .then((cookie) => {
+    // cookie is an object with "domain", "name" and other properties
+  })
+```
+
+You can check the cookie existence without comparing any of its properties
+
+```javascript
+cy.getCookie('my-session-cookie').should('exist')
+```
+
+If you need the cookie value, for example to use in a subsequent call
+
+```javascript
+let cookie
+
+cy.getCookie('session_id')
+  .should('exist')
+  .then((c) => {
+    // save cookie until we need it
+    cookie = c
+  })
+// some time later, force the "cy.request"
+// to run ONLY after the cookie has been set
+// by placing it inside ".then"
+cy.wrap(null).then(() => {
```

Comment:
Is this really the best advice we have? cy.wrap(null)? 😥

Reply:
actually you are right - this is no longer necessary, I can simply do `cy.then`

Reply:
I don't really think you're supposed to do `cy.then()` either. 😬We have an issue open to convert `.then()` to a child command - rendering this broken https://github.com/cypress-io/cypress/issues/466

Can't you just add in the example something arbitrary to chain it off of? Like... `cy.get('#submit').click().then(...`

Reply:
reworked it one more time